presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -225,7 +230,20 @@ private Table getTable(SemiTransactionalHiveMetastore metastore, SchemaTableName
                     filter.add(HivePartitionKey.HIVE_DEFAULT_DYNAMIC_PARTITION);
                 }
                 else if (value instanceof Slice) {
-                    filter.add(((Slice) value).toStringUtf8());
+                    Type type = domain.getType();
```

Comment:
This change looks good, but I am not quite sure why it work e.g. for short decimal.
It uses `long` as container type so for value `3.14` value`"314"` will be generated as filter.
I guess Hive expects `"3.14"`. 

Note: This hole dispatching based on container type looks like wrong approach. We should do the dispatching based on the Presto type. That is what you did within `Slice` branch.